src/core/cardano/walletConnect/peerConnection.test.ts

```diff
@@ -0,0 +1,153 @@
+import { DataType, SecureStorage } from "@aparajita/capacitor-secure-storage";
+import { IdentityWalletConnect } from "./identityWalletConnect";
+import { Agent } from "../../agent/agent";
+import {
+  PeerConnectionMetadataRecord,
+  PeerConnectionStorage,
+} from "../../agent/records";
+import { PeerConnection } from "./peerConnection";
+import { KeyStoreKeys } from "../../storage";
+require("fake-indexeddb/auto");
```

Comment:
Why do we need this? I would have thought we would mock everything related to Meerkat so it would never write to indexeddb?

Reply:
When the code executes `new Meerkat`, the error will appear because the environment is different, so it would be pretty hard to create its mock.

Reply:
OK but in this case, does this mean we are starting up an actual Meerkat instance? which could be quite "heavy" for a unit test

Reply:
Yes. The `Meerkat` is a `protected` inside `CardanoPeerConnect` that the `IdentityWalletConnect` is extending from.

Reply:
Can we mock `CardanoPeerConnect` or maybe use `spyOn` to stop it from creating meerkat instances?

Reply:
![image](https://github.com/cardano-foundation/cf-identity-wallet/assets/142211701/3e9dd669-d5a6-4c35-96fe-6309869f1880)
The function that calls `new Meerkat` is called inside the CardanoPeerConnect's constructor so it would be more complicated than using a `fake-indexeddb`.

Reply:
But can we not spyOn `setUpDsicoveryMeerkat`?

actually though - can't we just mock the meerkat library completely? (`@fabianbormann/meerkat`)

Reply:
`setUpDiscoveryMeerkat` is a protected function so we can't use `spyOn` with it. Since `Meerkat` is also extending from `EventEmitter` so it will be very complex to mock it entirely.

Reply:
agreed to leave this as is and possibly convert this into more full fledged integration tests in another ticket